### PR TITLE
Add best practises

### PR DESCRIPTION
**Pull request by wagnerand**
from https://docs.google.com/document/d/1EFE6beSAxcR5SHhx6nrNMjQiLBeIrohyEZD5B1FMsl8

@andymckay r?

### CODE REVIEW

**Comment by tofumatt** [comment]
r+ from me.

File: reference/git_github.rst [hunk 1]
```diff
@@ -3,6 +3,16 @@ Git and Github
 
 This document describes some best practices for using Git and Github.
 
+Best practises
+--------------
+- Create a separate pull request for each bug. If you have multiple commits in one pull request, squash them into one.
```
Comment (erikrose) [inline reference/git_github.rst]: This depends on how complicated the change is. Complex changes can be easier to review as a series of small, self-contained commits.

Comment (wagnerand) [inline reference/git_github.rst]: While true, beginners do not work on complex changes and except for the case you described, I think it's fair to say that we want to have one commit only.

Comment (erikrose) [inline reference/git_github.rst]: The Reference section (of which this is part) is currently aimed at all contribs, not just beginners. From the front page:

> If you’re already a Webdev contributor, you can find generally-useful info, such as style guides, in the Reference.

**Comment by wagnerand** [comment]
@erikrose I clarified that line.

**Comment by erikrose** [comment]
Thanks! That line looks fine now.

**Comment by wagnerand** [comment]
Great, would someone mind merging this, please?

File: reference/git_github.rst [hunk 2]
```diff
@@ -3,6 +3,17 @@ Git and Github
 
 This document describes some best practices for using Git and Github.
 
+Best practises
+--------------
+- Create a separate pull request for each bug.
+- If you have multiple commits in one pull request, squash them into one. For complex patches, having multiple commits might make the review easier.
+- If you are working with multiple bugs at a time, make sure you create a separate branches for each bug.
+- If you need to make changes to the commit, perhaps after feedback, amend the original commit and comment in the pull request what you've changed.  Don't create a new pull request or commit.
+- Include  the bug number in your commit & pull request title. If possible  follow this style -`Some text description (bug XXXXXXX)`.
```
Comment (andymckay) [inline reference/git_github.rst]: That makes sense for bugzilla bugs, for github issues, maybe not. For example there is an internal way in github of referencing an issue in a pull request.

Comment (tofumatt) [inline reference/git_github.rst]: Fair enough; how is #54?

**Pull request merged**